frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/stateUpdate/vmSettings/storage-tab-state-update.ts

```diff
@@ -46,11 +59,28 @@ export const prefillInitialDiskUpdater = ({ id, prevState, dispatch, getState }:
         );
       }
     } else {
-      dispatch(
-        vmWizardInternalActions[InternalActionType.UpdateStorage](id, {
-          id: oldSourceStorage ? oldSourceStorage.id : getNextIDResolver(getStorages(state, id))(),
-          ...newSourceStorage,
-        }),
+      if (newType !== oldType) {
+        dispatch(
+          vmWizardInternalActions[InternalActionType.UpdateStorage](id, {
+            id: oldSourceStorage
+              ? oldSourceStorage.id
+              : getNextIDResolver(getStorages(state, id))(),
+            ...newSourceStorage,
+          }),
+        );
+        return;
+      }
+      [
+        VMWizardStorageType.PROVISION_SOURCE_DISK,
+        VMWizardStorageType.WINDOWS_GUEST_TOOLS,
+      ].map((storageType) =>
+        dispatch(
```

Comment:
we should first check if the disk has valid bus and only if it doesn't, set the default one. The user already might have chosen a desired option.